diff --git a/node_modules/@ledgerhq/react-native-hid/android/build.gradle b/node_modules/@ledgerhq/react-native-hid/android/build.gradle
index d6eebd0..40aac04 100644
--- a/node_modules/@ledgerhq/react-native-hid/android/build.gradle
+++ b/node_modules/@ledgerhq/react-native-hid/android/build.gradle
@@ -21,11 +21,10 @@ allprojects {
 apply plugin: 'com.android.library'
 
 android {
-  compileSdkVersion 29
-
   defaultConfig {
     minSdkVersion 24
-    targetSdkVersion 29
+    compileSdkVersion 33
+    targetSdkVersion 33
     versionCode 1
     versionName "1.0"
   }
diff --git a/node_modules/@ledgerhq/react-native-hid/android/src/main/java/com/ledgerwallet/hid/ReactHIDModule.java b/node_modules/@ledgerhq/react-native-hid/android/src/main/java/com/ledgerwallet/hid/ReactHIDModule.java
index 26b4134..5edd1b5 100644
--- a/node_modules/@ledgerhq/react-native-hid/android/src/main/java/com/ledgerwallet/hid/ReactHIDModule.java
+++ b/node_modules/@ledgerhq/react-native-hid/android/src/main/java/com/ledgerwallet/hid/ReactHIDModule.java
@@ -10,6 +10,7 @@ import android.hardware.usb.UsbDeviceConnection;
 import android.hardware.usb.UsbEndpoint;
 import android.hardware.usb.UsbInterface;
 import android.hardware.usb.UsbManager;
+import android.os.Build;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.Promise;
@@ -162,10 +163,18 @@ public class ReactHIDModule extends ReactContextBaseJavaModule {
         return result.toByteArray();
     }
 
+    public static PendingIntent createPendingIntentGetBroadcast(Context context, int id, Intent intent, int flag) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            return PendingIntent.getBroadcast(context, id, intent, PendingIntent.FLAG_MUTABLE | flag);
+        } else {
+            return PendingIntent.getBroadcast(context, id, intent, flag);
+        }
+    }
+
     private void requestUsbPermission(UsbManager manager, UsbDevice device, Promise p) {
         try {
             ReactApplicationContext rAppContext = getReactApplicationContext();
-            PendingIntent permIntent = PendingIntent.getBroadcast(rAppContext, 0, new Intent(ACTION_USB_PERMISSION), 0);
+            PendingIntent permIntent = createPendingIntentGetBroadcast(rAppContext, 0, new Intent(ACTION_USB_PERMISSION), 0);
             registerBroadcastReceiver(p);
             manager.requestPermission(device, permIntent);
         } catch (Exception e) {
