diff --git a/node_modules/expo-camera/ios/Current/BarcodeScanner.swift b/node_modules/expo-camera/ios/Current/BarcodeScanner.swift
index 94f2025..19d2b8d 100644
--- a/node_modules/expo-camera/ios/Current/BarcodeScanner.swift
+++ b/node_modules/expo-camera/ios/Current/BarcodeScanner.swift
@@ -4,7 +4,7 @@ import AVFoundation
 let BARCODE_TYPES_KEY = "barcodeTypes"
 
 actor BarcodeScanner: NSObject, BarcodeScanningResponseHandler {
-  var onBarcodeScanned: (([String: Any]?) -> Void)?
+  private var onBarcodeScanned: (([String: Any]?) -> Void)?
   var isScanningBarcodes = false
 
   // MARK: - Properties
@@ -76,6 +76,10 @@ actor BarcodeScanner: NSObject, BarcodeScanningResponseHandler {
     }
   }
 
+  func setOnBarcodeScanned(_ onBarcodeScanned: @escaping ([String: Any]?) -> Void) {
+    self.onBarcodeScanned = onBarcodeScanned
+  }
+
   func maybeStartBarcodeScanning() async {
     guard isScanningBarcodes else {
       return
diff --git a/node_modules/expo-camera/ios/Current/CameraView.swift b/node_modules/expo-camera/ios/Current/CameraView.swift
index 2694c66..4064c0b 100644
--- a/node_modules/expo-camera/ios/Current/CameraView.swift
+++ b/node_modules/expo-camera/ios/Current/CameraView.swift
@@ -792,13 +792,16 @@ public class CameraView: ExpoView, EXAppLifecycleListener,
 
   private func createBarcodeScanner() -> BarcodeScanner {
     let scanner = BarcodeScanner(session: session, sessionQueue: sessionQueue)
-    scanner.setPreviewLayer(layer: previewLayer)
-    scanner.onBarcodeScanned = { [weak self] body in
-      guard let self else {
-        return
-      }
-      if let body {
-        self.onBarcodeScanned(body)
+    
+    Task {
+      await scanner.setPreviewLayer(layer: previewLayer)
+      await scanner.setOnBarcodeScanned { [weak self] body in
+        guard let self else {
+          return
+        }
+        if let body {
+          self.onBarcodeScanned(body)
+        }
       }
     }
 
