diff --git a/node_modules/react-native-reanimated/lib/module/propsAllowlists.js b/node_modules/react-native-reanimated/lib/module/propsAllowlists.js
index 6996a02..a6f7ecb 100644
--- a/node_modules/react-native-reanimated/lib/module/propsAllowlists.js
+++ b/node_modules/react-native-reanimated/lib/module/propsAllowlists.js
@@ -121,7 +121,8 @@ export const PropsAllowlists = {
     color: true,
     tintColor: true,
     shadowColor: true,
-    placeholderTextColor: true
+    placeholderTextColor: true,
+    text: true
   }
 };
 //# sourceMappingURL=propsAllowlists.js.map
\ No newline at end of file
