diff --git a/node_modules/react-native-reanimated/src/ConfigHelper.ts b/node_modules/react-native-reanimated/src/ConfigHelper.ts
index fac2429..7d63384 100644
--- a/node_modules/react-native-reanimated/src/ConfigHelper.ts
+++ b/node_modules/react-native-reanimated/src/ConfigHelper.ts
@@ -117,6 +117,7 @@ let NATIVE_THREAD_PROPS_WHITELIST: Record<string, boolean> = {
   tintColor: true,
   shadowColor: true,
   placeholderTextColor: true,
+  text: true
 };
 
 function configureProps(): void {
