diff --git a/node_modules/react-native-vision-camera/ios/CameraView.swift b/node_modules/react-native-vision-camera/ios/CameraView.swift
index 93a9f7c..8f91684 100644
--- a/node_modules/react-native-vision-camera/ios/CameraView.swift
+++ b/node_modules/react-native-vision-camera/ios/CameraView.swift
@@ -170,6 +170,7 @@ public final class CameraView: UIView {
     NotificationCenter.default.removeObserver(self,
                                               name: UIDevice.orientationDidChangeNotification,
                                               object: nil)
+    self.captureSession.stopRunning()
   }
 
   override public func willMove(toSuperview newSuperview: UIView?) {
