diff --git a/node_modules/react-native-screens/ios/RNSScreen.mm b/node_modules/react-native-screens/ios/RNSScreen.mm
index d6975f7..f12eaee 100644
--- a/node_modules/react-native-screens/ios/RNSScreen.mm
+++ b/node_modules/react-native-screens/ios/RNSScreen.mm
@@ -939,11 +939,6 @@ - (void)viewDidLayoutSubviews
   if (isDisplayedWithinUINavController || isPresentedAsNativeModal) {
 #ifdef RCT_NEW_ARCH_ENABLED
     [self.screenView updateBounds];
-#else
-    if (!CGRectEqualToRect(_lastViewFrame, self.screenView.frame)) {
-      _lastViewFrame = self.screenView.frame;
-      [((RNSScreenView *)self.viewIfLoaded) updateBounds];
-    }
 #endif
   }
 }
