diff --git a/node_modules/react-native-reanimated/ios/REANodesManager.m b/node_modules/react-native-reanimated/ios/REANodesManager.m
index fda5bb2..8d1e0c5 100644
--- a/node_modules/react-native-reanimated/ios/REANodesManager.m
+++ b/node_modules/react-native-reanimated/ios/REANodesManager.m
@@ -297,8 +297,9 @@ - (void)performOperations
     }
 
     if (_mounting) {
-      _mounting();
+      volatile void (^_mountingPtr)(void) = _mounting;
       _mounting = nil;
+      _mountingPtr();
     }
   }
   _wantRunUpdates = NO;
